scripts/prepare-preview-manifest.sh

```diff
@@ -18,13 +18,19 @@ shorthash="$1"
 # of the version. Technically we'd want to bump the non-prerelease portion as
 # well if we wanted this to be SemVer-compliant, but it was simpler not to.
 # This is just for testing, it doesn't need to strictly follow SemVer.
-jq --raw-output ".version |= split(\"-\")[0] + \"-preview.${shorthash}\"" ./package.json > temp.json
+jq --raw-output --arg hash "$shorthash" '.version |= split("-")[0] + "-preview.\($hash)"' ./package.json |
+
+# The workspace dependencies are updated to point to the exact preview build
+# version, so that it does not try to resolve a different version from the
+# registry.
+jq --raw-output 'if has("devDependencies") then .devDependencies |= map_values(if . == "workspace:^" then "workspace:*" else . end) else . end' |
```

Comment:
Totally agreed, though I'd still prefer to merge this first and address this later. This script is duplicated across three repositories, so it's a little inconvenient to make changes to it.

Reply:
Actually I'll do it now here at least, so that we don't forget: https://github.com/MetaMask/core/pull/1445/commits/686dacf755826012c5a3ea13380241952683a075